compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -2041,6 +2041,10 @@ class TypeComparer(@constructorOnly initctx: Context) extends ConstraintHandling
     else if (!tp2.exists) tp1
     else if tp1.isAny && !tp2.isLambdaSub || tp1.isAnyKind || isBottom(tp2) then tp2
     else if tp2.isAny && !tp1.isLambdaSub || tp2.isAnyKind || isBottom(tp1) then tp1
+    else if tp1.containsFromJavaObject && !tp2.isLambdaSub then
+      tp2 // If tp1 is FromJavaObject or a union containing FromJavaObject, tp2 <:< tp1
```

Comment:
I don't quite understand what the changes in glb/lub are for, I thought we were trying to preserve FromJavaObject but it seems like this would discard them more often? If those are not needed for any test case I would drop them and keep just the changes in TypeBounds#& which LGTM.